storage/writer.go

```diff
@@ -107,12 +107,25 @@ type Writer struct {
 	// Append is a parameter to indicate whether the writer should use appendable
 	// object semantics for the new object generation. Appendable objects are
 	// visible on the first Write() call, and can be appended to until they are
-	// finalized. The object is finalized on a call to Close().
+	// finalized. If Writer.FinalizeOnClose is set to true, the object is finalized
+	// when Writer.Close() is called; otherwise, the object is left unfinalized
+	// and can be appended to later.
 	//
 	// Append is only supported for gRPC. This feature is in preview and is not
 	// yet available for general use.
 	Append bool
 
+	// FinalizeOnClose indicates whether the writer should finalize an object when
+	// closing the write stream. This only applies to Writers where Append is
+	// true, since append semantics allow a prefix of the object to be durable and
+	// readable. By default, objects written with Append semantics will not be
```

Comment:
nit: not sure the audience for this, but I would specify here something to the effect that the "prefix of the object" is the first N bytes or bytes uploaded so far; at least personally I'd never seen this term used this way outside of this and figure it might be easily confusable for object name prefix.